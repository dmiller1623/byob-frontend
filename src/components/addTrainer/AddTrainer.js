import React, { Component } from 'react'
import './addTrainer.css'

class AddTrainer extends Component {
  constructor(props) {
    super();
    this.state = {
      name: '',
      level: 1
    }
  }

  handleChange = (e) => {
    const { value } = e.target;
    this.setState({
      name: value
    })
  } 

  handleSubmit = () => {
    this.props.addTrainer(this.state)
  }


  render() {
    return (
      <div className='add-trainer-input'>
        <div className='image-section'>
          <img
            className='trainer-image'
            src='https://vignette.wikia.nocookie.net/pokemon-and-digimon/images/7/7a/Hoennash.png/revision/latest?cb=20130701183623'
            alt='ash ketchum'
            />
        </div>
        <div className='input-section'>
          <input
            placeholder='Enter Trainer Name'
            className='trainer-input'
            onChange={(e) => this.handleChange(e)}
            type='text'
            value={this.state.name}
          />
          <button className='add-trainer-buttons' onClick={this.handleSubmit}>Add trainer</button>
          <button className='add-trainer-buttons' onClick={this.props.toggleAddTrainer}>Go Back</button>
        </div>
      </div>
    )
  }
}

export default AddTrainer