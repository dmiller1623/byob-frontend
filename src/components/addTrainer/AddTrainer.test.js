import React from 'react';
import { shallow } from 'enzyme'; 
import AddTrainer from './AddTrainer.js';

describe('AddTrainer', () => {
  let wrapper
 
  it('should match the snapshot', () => {
    wrapper = shallow(<AddTrainer />);
    expect(wrapper).toMatchSnapshot();
  });
});