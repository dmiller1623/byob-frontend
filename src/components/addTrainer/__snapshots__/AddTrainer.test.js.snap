// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTrainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddTrainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="image-section"
        >
          <img
            alt="ash ketchum"
            className="trainer-image"
            src="https://vignette.wikia.nocookie.net/pokemon-and-digimon/images/7/7a/Hoennash.png/revision/latest?cb=20130701183623"
          />
        </div>,
        <div
          className="input-section"
        >
          <input
            className="trainer-input"
            onChange={[Function]}
            placeholder="Enter Trainer Name"
            type="text"
            value=""
          />
          <button
            className="add-trainer-buttons"
            onClick={[Function]}
          >
            Add trainer
          </button>
          <button
            className="add-trainer-buttons"
          >
            Go Back
          </button>
        </div>,
      ],
      "className": "add-trainer-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="ash ketchum"
            className="trainer-image"
            src="https://vignette.wikia.nocookie.net/pokemon-and-digimon/images/7/7a/Hoennash.png/revision/latest?cb=20130701183623"
          />,
          "className": "image-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "ash ketchum",
            "className": "trainer-image",
            "src": "https://vignette.wikia.nocookie.net/pokemon-and-digimon/images/7/7a/Hoennash.png/revision/latest?cb=20130701183623",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="trainer-input"
              onChange={[Function]}
              placeholder="Enter Trainer Name"
              type="text"
              value=""
            />,
            <button
              className="add-trainer-buttons"
              onClick={[Function]}
            >
              Add trainer
            </button>,
            <button
              className="add-trainer-buttons"
            >
              Go Back
            </button>,
          ],
          "className": "input-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "trainer-input",
              "onChange": [Function],
              "placeholder": "Enter Trainer Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add trainer",
              "className": "add-trainer-buttons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add trainer",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Go Back",
              "className": "add-trainer-buttons",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Go Back",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image-section"
          >
            <img
              alt="ash ketchum"
              className="trainer-image"
              src="https://vignette.wikia.nocookie.net/pokemon-and-digimon/images/7/7a/Hoennash.png/revision/latest?cb=20130701183623"
            />
          </div>,
          <div
            className="input-section"
          >
            <input
              className="trainer-input"
              onChange={[Function]}
              placeholder="Enter Trainer Name"
              type="text"
              value=""
            />
            <button
              className="add-trainer-buttons"
              onClick={[Function]}
            >
              Add trainer
            </button>
            <button
              className="add-trainer-buttons"
            >
              Go Back
            </button>
          </div>,
        ],
        "className": "add-trainer-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="ash ketchum"
              className="trainer-image"
              src="https://vignette.wikia.nocookie.net/pokemon-and-digimon/images/7/7a/Hoennash.png/revision/latest?cb=20130701183623"
            />,
            "className": "image-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "ash ketchum",
              "className": "trainer-image",
              "src": "https://vignette.wikia.nocookie.net/pokemon-and-digimon/images/7/7a/Hoennash.png/revision/latest?cb=20130701183623",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="trainer-input"
                onChange={[Function]}
                placeholder="Enter Trainer Name"
                type="text"
                value=""
              />,
              <button
                className="add-trainer-buttons"
                onClick={[Function]}
              >
                Add trainer
              </button>,
              <button
                className="add-trainer-buttons"
              >
                Go Back
              </button>,
            ],
            "className": "input-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "trainer-input",
                "onChange": [Function],
                "placeholder": "Enter Trainer Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add trainer",
                "className": "add-trainer-buttons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add trainer",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Go Back",
                "className": "add-trainer-buttons",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Go Back",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
